File: 0274-h-index/0274-h-index.py
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        # Sort the citations array in descending order
        citations.sort(reverse=True)
        
        # Iterate through the sorted citations array
        h = 0
        for i in range(len(citations)):
            # If the number of citations is greater than or equal to the index + 1 (h), update h
            if citations[i] >= i + 1:
                h = i + 1
            else:
                break
        
        return h

File: 0274-h-index/0274-h-index.java
class Solution {
    public int hIndex(int[] citations) {
        // Convert int[] to Integer[] to use Collections.reverseOrder()
        Integer[] citationsArray = Arrays.stream(citations).boxed().toArray(Integer[]::new);

        // Sort the citations array in descending order
        Arrays.sort(citationsArray, Collections.reverseOrder());

        // Iterate through the sorted citations array to calculate the h-index
        int h = 0;
        for (int i = 0; i < citationsArray.length; i++) {
            if (citationsArray[i] >= i + 1) {
                h = i + 1;
            } else {
                break;
            }
        }

        return h;
        
    }
}